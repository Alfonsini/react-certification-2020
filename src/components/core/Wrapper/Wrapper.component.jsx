import React from 'react';

import { StyledWrapper } from './Wrapper.styles';

function Wrapper({ children }) {
  return <StyledWrapper role="main">{children}</StyledWrapper>;
}

export default Wrapper;
