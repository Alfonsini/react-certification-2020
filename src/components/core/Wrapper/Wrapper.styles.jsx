import styled from 'styled-components';

const StyledWrapper = styled.main`
  position: fixed;
  left: 0;
  height: 100%;

  margin-left: 85px;
  margin-right: 0px;
  margin-top: 10px;

  width: 100%;
  background-color: ${(props) => `${props.theme.backgroundColorWrapper}`};
  box-shadow: 0 4px 24px 0 rgb(34 41 47 / 10%);
  border-radius: 0.428rem;

  width: calc(100% - 95px);
  height: calc(100% - 20px);

  @media screen and (max-width: 500px) {
    width: calc(100% - 20px);
    margin-left: 10px;
  }
`;

export { StyledWrapper };
