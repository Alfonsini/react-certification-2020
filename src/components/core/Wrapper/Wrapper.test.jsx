import React from 'react';
import { render, screen } from '@testing-library/react';

import Wrapper from '.';

beforeEach(() => {});

afterEach(() => {});

describe('Wrapper', () => {
  test('wrapper render correctly', () => {
    render(<Wrapper />);
    const aside = screen.getByRole('main');
    expect(aside).toMatchSnapshot();
  });
});
