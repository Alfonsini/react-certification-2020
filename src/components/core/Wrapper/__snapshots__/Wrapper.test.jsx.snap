// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wrapper wrapper render correctly 1`] = `
<main
  class="sc-bdfBwQ fMDArA"
  role="main"
/>
`;
